### PR TITLE
fix: Corrects potential out of bounds error for the file list prompt when the list overflows

### LINKED ISSUE #19: Bug: Files prompt is not properly bounded when files overflow
Currently if the files prompt is truncated ("and x more") the lower bound for this list is not where it's meant to be. The lower bound currently relies upon checking the total number of options, not necessarily the total amount of options displayed. Going past the truncation message will cause a crash as the index will be out of bounds. 

I'll try to push a PR through at some point

https://user-images.githubusercontent.com/33403762/109992739-2ecb0c00-7cd1-11eb-8265-9125fa6f8bb2.mp4

### PR DESCRIPTION
**Pull request by rschristian**
Closes #19

### CODE REVIEW

File: src/prompt/files_prompt.rs [hunk 1]
```diff
@@ -42,6 +42,11 @@ impl<'a> FilesPrompt<'a> {
             .get_figlet()
             .expect("Ensure figlet_file points to a valid file, or remove it.");
 
+        // Padded limit (never overflows by 1 item)
+        let total = self.options.len();
+        let max = 15;
+        let take = if total > max { max - 2 } else { total + 1 };
```
Comment (brigand) [inline src/prompt/files_prompt.rs:48]: It seems strange that it jumps back down after passing the threshold. Is this intended?

- total = 14, then take = 15
- total = 15, then take = **16**
- total = 16, then take = **13**
- total = 17, then take = 13

Comment (rschristian) [reply src/prompt/files_prompt.rs:48]: That's what was already happening? I only made slight edits to the values. I can take a look at that though.

I'd like to swap out the static max to depend on the terminal height as well at some point.

File: src/prompt/files_prompt.rs [hunk 2]
```diff
@@ -134,16 +139,12 @@ impl<'a> FilesPrompt<'a> {
                     };
                 }
                 Some((KeyCode::Down, _, _, true)) => {
-                    let total = self.options.len() as u16 + 1;
-
-                    self.focused_index += total.saturating_sub(1);
+                    self.focused_index += (take as u16 + 1).saturating_sub(1);
```
Comment (brigand) [inline src/prompt/files_prompt.rs:142]: `(take as u16 + 1).saturating_sub(1)` is the same as `take as u16` here.

File: src/prompt/files_prompt.rs [hunk 3]
```diff
@@ -134,16 +139,12 @@ impl<'a> FilesPrompt<'a> {
...
                 }
                 Some((KeyCode::Down, _, _, false)) => {
-                    let total = self.options.len() as u16 + 1;
-
                     self.focused_index += 1;
-                    if self.focused_index >= total {
-                        self.focused_index = total.saturating_sub(1);
+                    if self.focused_index >= take as u16 + 1 {
+                        self.focused_index = (take as u16 + 1).saturating_sub(1);
```
Comment (brigand) [inline src/prompt/files_prompt.rs:147]: Same as previous comment. Simply `take as u16`

**Review by brigand — CHANGES_REQUESTED** [review]

**Comment by brigand** [comment]
I'm a bit unclear on how this is solving the issue. Could you explain?

File: src/prompt/files_prompt.rs (hunk 2 above)
Comment (rschristian) [reply src/prompt/files_prompt.rs]: Oh, duh, apologies.

**Comment by rschristian** [comment]
@brigand 

Previously the bound was defined by the options length. If we had 50 options, aka files, the lower bound would be set at 50. So you could just go past.

The error would throw then from `render_one_line`. You could go to index say 25 of the options, as your bound was 50, but only 25 lines were actually rendered. There was no index 25 in terms of lines rendered.

**Review by brigand — APPROVED** [review]

**Pull request merged**